test/test_primal_dual_objective.py

```diff
@@ -0,0 +1,50 @@
+from SurfaceTopography import make_sphere
+import ContactMechanics as Solid
+import scipy.optimize as optim
+
+import numpy as np
+
+nx, ny = 128,128
```

Comment:
Not sure if it is a good practice to define variables outside the test. One would rather use fixtures for this. Or just copy this code inside the test function.